cmd/tsbs_generate_queries/databases/siridb/devops_test.go

```diff
@@ -94,3 +341,74 @@ func TestDevopsFillInQuery(t *testing.T) {
 		t.Errorf("Wrong query: got %s want %s", got, siriql)
 	}
 }
+
+type Case struct {
+	desc               string
+	input              int
+	fail               bool
+	failMsg            string
+	expectedHumanLabel string
+	expectedHumanDesc  string
+	expectedQuery      string
+}
+
+func runTestCases(t *testing.T, testFunc func(*Devops, Case) query.Query, s time.Time, e time.Time, cases []Case) {
+	rand.Seed(123) // Setting seed for testing purposes.
+
+	for _, c := range cases {
+		t.Run(c.desc, func(t *testing.T) {
+			var wg sync.WaitGroup
+			d := NewDevops(s, e, 10)
+
+			if c.fail {
+				f := devops.GetFatal()
+				devops.SetFatal(func(wg *sync.WaitGroup) func(string, ...interface{}) {
+					return func(s string, v ...interface{}) {
+						defer wg.Done()
+						if s != c.failMsg {
+							t.Fatalf("incorrect fail message: got %s, want %s", s, c.failMsg)
+						}
+						runtime.Goexit()
```

Comment:
moot point if we go with my other pr (#77), but is this necessary? we don't have to have the same semantics of log.Fatalf (i.e. killing the process). If we take this out, we can reduce the amount of defer/waitgroup also, making this an easier func to understand

Reply:
I agree but if we don't kill the process, fatal will be called multiple times in a test case and thats something that the existing function stops from happening. That means tests would fail and I would have to have multiple `fatalMsg`s to check each one.

I know it looks over-engineered but I didn't find a better way of replicating the current behavior in order that the tests reflect real-world usage.